packages/axelar-soroban-std/src/types.rs

```diff
@@ -1,3 +1,10 @@
-use soroban_sdk::BytesN;
+use soroban_sdk::{contracttype, Address, BytesN};
 
 pub type Hash = BytesN<32>;
+
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct TokenDetails {
+    pub token_addr: Address,
+    pub amount: i128,
```

Comment:
```suggestion
pub struct Token {
    pub address: Address, // TODO: check if this can be changed to a TokenClient type instead which is richer than Address, or a generic type implementing TokenInterface
    pub amount: i128,
```
best to avoid truncated names. `token.address`, `token.amount` usage should be more readable